client/packages/common/src/ui/components/navigation/AppNavLink/AppNavLink.tsx

```diff
@@ -99,27 +109,46 @@ export const AppNavLink: FC<AppNavLinkProps> = props => {
     drawer.onClick();
   };
 
+  const showConfirmation = useConfirmationModal({
+    onConfirm: handleClick,
+    message: confirmMessage || t('messages.confirm-delete-generic'),
+    title: confirmTitle || t('heading.are-you-sure'),
+  });
+
   const CustomLink = React.useMemo(
     () =>
-      React.forwardRef<HTMLAnchorElement>((linkProps, ref) =>
-        !end && !!inactive ? (
-          <span
-            {...linkProps}
-            onClick={() => drawer.onExpand(to)}
-            data-testid={`${to}_hover`}
-          />
-        ) : (
-          <Link
-            {...linkProps}
-            ref={ref}
-            to={to}
-            role="link"
-            aria-label={text}
-            title={text}
-            onClick={handleClick}
-          />
-        )
-      ),
+      React.forwardRef<HTMLAnchorElement>((linkProps, ref) => {
+        if (!end && !!inactive) {
+          return (
+            <span
+              {...linkProps}
+              onClick={() => drawer.onExpand(to)}
+              data-testid={`${to}_hover`}
+            />
+          );
+        } else if (requireConfirmation) {
+          return (
+            <span
+              {...linkProps}
+              onClick={
+                showConfirmation as React.MouseEventHandler<HTMLSpanElement>
```

Comment:
I am handling type incompatibility by defining this type here. But am not confident in that this is right.

Reply:
yeah - I would normally use the `onClick={() => showConfirmation()}` syntax in a case like this - it means that typescript can infer the types and makes things easier 😉 

also agree with @CarlosNZ recommendation to lift the logic out if you can, that would be neater.